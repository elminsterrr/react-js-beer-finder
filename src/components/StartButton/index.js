import React, { Component } from 'react';
import RaisedButton from 'material-ui/RaisedButton';

import './style.css';

class StartButton extends Component {
  constructor() {
    super();

    this.state = {
      startWasClicked: false,
    };
  }

  handleStartButton = () => {
    this.setState({ startWasClicked: true });
  };

  beerListingView() {
    if (this.state.startWasClicked) {
      return <div>YES! It was clicked!</div>;
    }
    // Else return just single <div />
    return <div />;
  }

  render() {
    return (
      <div>
        <div className="StartButton-container">
          {!this.state.startWasClicked && (
            <RaisedButton
              label="Start Here"
              className="StartButton-main"
              onClick={this.handleStartButton}
            />
          )}
        </div>
        {this.beerListingView()}
      </div>
    );
  }
}

export default StartButton;
