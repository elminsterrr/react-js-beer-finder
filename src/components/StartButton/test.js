import React from 'react';
import { shallow } from 'enzyme';
import StartButton from './';

describe('StartButton', () => {
  const wrapper = shallow(<StartButton />);

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders `StartButton-container` children correctly', () => {
    expect(wrapper.find('.StartButton-container').children().length).toEqual(1);
  });

  it('changes `state().startWasClicked` to `true` when clicking the `Start Here` button', () => {
    wrapper.find('.StartButton-main').simulate('click');

    expect(wrapper.state().startWasClicked).toBeTruthy();
  });
});
