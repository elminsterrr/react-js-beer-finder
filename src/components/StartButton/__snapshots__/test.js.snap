// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartButton renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StartButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="StartButton-container"
>
          <RaisedButton
                    className="StartButton-main"
                    disabled={false}
                    fullWidth={false}
                    label="Start Here"
                    labelPosition="after"
                    onClick={[Function]}
                    primary={false}
                    secondary={false}
          />
</div>,
        <div />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RaisedButton
            className="StartButton-main"
            disabled={false}
            fullWidth={false}
            label="Start Here"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
/>,
          "className": "StartButton-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "StartButton-main",
            "disabled": false,
            "fullWidth": false,
            "label": "Start Here",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": false,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="StartButton-container"
>
            <RaisedButton
                        className="StartButton-main"
                        disabled={false}
                        fullWidth={false}
                        label="Start Here"
                        labelPosition="after"
                        onClick={[Function]}
                        primary={false}
                        secondary={false}
            />
</div>,
          <div />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RaisedButton
              className="StartButton-main"
              disabled={false}
              fullWidth={false}
              label="Start Here"
              labelPosition="after"
              onClick={[Function]}
              primary={false}
              secondary={false}
/>,
            "className": "StartButton-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "StartButton-main",
              "disabled": false,
              "fullWidth": false,
              "label": "Start Here",
              "labelPosition": "after",
              "onClick": [Function],
              "primary": false,
              "secondary": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
